kolibri/core/public/api.py

```diff
@@ -117,3 +123,34 @@ def get_public_channel_lookup(request, version, identifier):
         json.dumps(PublicChannelSerializer(channel_list, many=True).data),
         content_type="application/json",
     )
+
+
+@csrf_exempt
+@gzip_page
+def get_public_file_checksums(request, version):
+    """ Endpoint: /public/<version>/file_checksums/ """
+    if version == "v1":
+        if request.content_type == "application/json":
+            data = request.body
+        elif request.content_type == "application/gzip":
+            with gzip.GzipFile(fileobj=io.BytesIO(request.body)) as f:
+                data = f.read()
+        else:
+            return HttpResponseBadRequest("POST body must be either json or gzip")
+        checksums = json.loads(data.decode("utf-8"))
+        available_checksums = set(
+            LocalFile.objects.filter(available=True, id__in=checksums)
+            .values_list("id", flat=True)
+            .distinct()
+        )
+        return HttpResponse(
+            sum(
+                int(checksum in available_checksums) << i
+                for i, checksum in enumerate(checksums)
```

Comment:
Perhaps put this in a helper function alongside `_generate_mask`, so their logic is clearly defined in parallel?

Reply:
Done